core/src/test/java/com/datastax/oss/driver/internal/core/loadbalancing/DefaultLoadBalancingPolicyQueryPlanTest.java

```diff
@@ -331,6 +331,30 @@ public void should_reorder_first_two_replicas_when_first_has_more_in_flight_than
     then(dsePolicy).should(never()).diceRoll1d4();
   }
 
+  @Test
+  public void should_not_shuffle_local_rack_replica_with_all_replicas_healthy() {
+    // Given
+    given(request.getRoutingKeyspace()).willReturn(KEYSPACE);
+    given(request.getRoutingKey()).willReturn(ROUTING_KEY);
+    given(tokenMap.getReplicas(KEYSPACE, null, ROUTING_KEY))
+        .willReturn(ImmutableSet.of(node1, node3, node5));
+    String localRack = "rack1";
+    given(dsePolicy.getLocalRack()).willReturn(localRack);
+    given(node1.getRack()).willReturn(localRack);
+    given(pool1.getInFlight()).willReturn(0);
+    given(pool3.getInFlight()).willReturn(0);
+    given(pool5.getInFlight()).willReturn(0);
+
+    // When
+    Queue<Node> plan1 = dsePolicy.newQueryPlan(request, session);
+    Queue<Node> plan2 = dsePolicy.newQueryPlan(request, session);
+
+    // Then
+    // nodes 1, 3 and 5 always first, round-robin on the rest
+    assertThat(plan1).containsExactly(node1, node3, node5, node2, node4);
+    assertThat(plan2).containsExactly(node1, node3, node5, node4, node2);
```

Comment:
Just to be sure, could you extend/write another test with another node being in the local rack? In this test the first replica happens to be in the local rack - maybe test a scenario where the second replica node is in the local rack.

Reply:
Done.